Source/Core/loadKTX.js

```diff
@@ -0,0 +1,273 @@
+         * @type Number
+         * @readonly
+         */
+        height : {
+            get : function() {
+                return this._height;
+            }
+        },
+        /**
+         * The compressed texture buffer.
+         * @type Uint8Array
+         * @readonly
+         */
+        bufferView : {
+            get : function() {
+                return this._buffer;
+            }
+        }
+    });
+
+    var fileIdentifier = [0xAB, 0x4B, 0x54, 0x58, 0x20, 0x31, 0x31, 0xBB, 0x0D, 0x0A, 0x1A, 0x0A];
+    var endiannessTest = 0x04030201;
+
+    var sizeOfUint32 = 4;
+
+    function parseKTX(data) {
+        var byteBuffer = new Uint8Array(data);
+
+        var isKTX = true;
+        for (var i = 0; i < fileIdentifier.length; ++i) {
+            if (fileIdentifier[i] !== byteBuffer[i]) {
+                isKTX = false;
+                break;
+            }
+        }
+
+        if (!isKTX) {
+            throw new RuntimeError('Invalid KTX file.');
+        }
+
+        var view;
+        var byteOffset;
+
+        if (defined(data.buffer)) {
+            view = new DataView(data.buffer);
+            byteOffset = data.byteOffset;
+        } else {
+            view = new DataView(data);
+            byteOffset = 0;
+        }
+
+        byteOffset += 12; // skip identifier
+
+        var endianness = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        if (endianness !== endiannessTest) {
+            throw new RuntimeError('File is the wrong endianness.');
+        }
+
+        var glType = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var glTypeSize = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var glFormat = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var glInternalFormat = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var glBaseInternalFormat = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var pixelWidth = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var pixelHeight = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var pixelDepth = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var numberOfArrayElements = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var numberOfFaces = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var numberOfMipmapLevels = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var bytesOfKeyValueByteSize = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        // skip metadata
+        byteOffset += bytesOfKeyValueByteSize;
+
+        var imageSize = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var texture = new Uint8Array(data, byteOffset, imageSize);
```

Comment:
While loading a glb with a compressed texture I noticed problems here due to `data` being a `Uint8Array` view of the glTF.

Changing it to `data.buffer` works.